src/Documenter.jl

```diff
@@ -750,4 +758,85 @@ function generate(pkgname::AbstractString; dir=nothing)
     nothing
 end
 
+const external_dir = "DOCUMENTER_EXTERNAL"
+
+"""
+A file outside of the documentation directory.
+"""
+struct ExternalPage
```

Comment:
This type could live in `Documents.jl` maybe. The `source` and `destination` functions as well.

Reply:
`ExternalPage` has been moved. I ended up getting rid of the other two functions because they became a lot simpler after some refactoring.